ui/pie.html

```diff
@@ -58,6 +77,52 @@
     })
     .open();
 
+// update logic runs whenever a new dropdown option is selected
+function update(val) {
+  if (valueLink) {
+    // close downlink on update
+    valueLink.close();
+
+    // update the node URI used in histogramLink to match the selected agent
+    agent_URI = document.getElementById('web_agent_select').value;
+    console.log(agent_URI);
+
+      // switch statement that changes the plot color according to web agent
+      switch(agent_URI) {
+        case "/unit/secondAgent":
+          color1 = "#7a85d6"
+          color2 = "#3c52f0";
+          color3 = "#182373";
+          break;
+        case "/unit/thirdAgent":
+          color1 = "#d5d9e0";
+          color2 = "#9da4b0"
+          color3 = "#6d7178";
+          break;
+        default:
+          // for master web agent
+          color1 = "#50e3c2";
+          color2 = "#359680";
+          color3 = "#1f5c4e"
+      }
+      sliceColors = [swim.Color.parse(color3),
+          swim.Color.parse(color2),
+          swim.Color.parse(color1)];
+
+          valueLink = swim.downlinkValue()
+              .hostUri("warp://localhost:9001")
+              .nodeUri(agent_URI)
+              .laneUri("latest")
+              .didSet(function (value) {
+                // TODO: change to stats
```

Comment:
I'm not sure if this `TODO` is still relevant, so I'll leave it up to you. Feel free to just take it out!